src/analysis/retail/warlock/demonology/modules/features/SummonDemonicTyrant.tsx

```diff
@@ -1,35 +1,66 @@
 import SPELLS from 'common/SPELLS';
 import TALENTS from 'common/TALENTS/warlock';
 import Analyzer, { Options, SELECTED_PLAYER, SELECTED_PLAYER_PET } from 'parser/core/Analyzer';
-import Events, { ApplyBuffEvent } from 'parser/core/Events';
+import Events, {
+  AnyEvent,
+  ApplyBuffEvent,
+  BeginCastEvent,
+  BeginChannelEvent,
+  EndChannelEvent,
+  GlobalCooldownEvent,
+} from 'parser/core/Events';
 import BoringSpellValueText from 'parser/ui/BoringSpellValueText';
 import Statistic from 'parser/ui/Statistic';
 import STATISTIC_ORDER from 'parser/ui/STATISTIC_ORDER';
-
 import DemoPets from '../pets/DemoPets';
 import './SummonTyrant.scss';
+import SpellLink from 'interface/SpellLink';
+import { QualitativePerformance } from 'parser/ui/QualitativePerformance';
+import ContextualSpellUsageSubSection from 'parser/core/SpellUsage/HideGoodCastsSpellUsageSubSection';
+import { ChecklistUsageInfo, SpellUse } from 'parser/core/SpellUsage/core';
+import { CastEvent } from 'parser/core/Events';
+import { combineQualitativePerformances } from 'common/combineQualitativePerformances';
+import Casts from 'interface/report/Results/Timeline/Casts';
+import EmbeddedTimelineContainer, {
+  SpellTimeline,
+} from 'interface/report/Results/Timeline/EmbeddedTimeline';
+import SpellUsable from 'parser/shared/modules/SpellUsable';
 
 const debug = false;
 
+const WILD_IMP_SPELL = 104317;
+
+interface TyrantCast {
+  event: CastEvent;
+  castNumber: number;
+  gfgOnCd: boolean;
+}
+
 class SummonDemonicTyrant extends Analyzer {
   static dependencies = {
     demoPets: DemoPets,
+    spellUsable: SpellUsable,
   };
 
   demoPets!: DemoPets;
+  spellUsable!: SpellUsable;
   _reignOfTyrannyPower!: number[];
   _hasReignOfTyranny!: boolean;
-  _petsPerCast!: Record<number, number>[];
+  _hasGFG!: boolean;
 
   _summsWithDemonicPower!: { [id: string]: any }[];
-  _tyrantCast!: number;
+  _tyrantsCast!: number;
+  _tyrantCasts!: TyrantCast[];
+  _castTimeline!: AnyEvent[];
 
   constructor(options: Options) {
     super(options);
-    this._tyrantCast = 0;
-    this._summsWithDemonicPower = [];
-    this._summsWithDemonicPower[0] = {};
+    this._tyrantsCast = 0;
```

Comment:
Analyzers that work on talents typically want to check if the player has that talent to avoid crashes and unnecessary analysis if the talent isn't taken.
```suggestion
    this.active = this.selectedCombatant.hasTalent(TALENTS.SUMMON_DEMONIC_TYRANT_TALENT);
    if (!this.active) {
      return;
    }
    this._tyrantsCast = 0;
```